wordpress-mobile/WordPress-Android
# [Deps] Migrate to Version Catalogs

Project Thread: paaHJt-7cv-p2

This PR migrates all dependencies and plugins in this project to [Version Catalogs](https://docs.gradle.org/current/userguide/platforms.html) and under the newly created [gradle/libs.versions.toml](https://github.com/wordpress-mobile/WordPress-Android/pull/21262/files#diff-697f70cdd88ba88fe77eebda60c7e143f6ad1286bca75017421e93ad84fb87df) file.

## Description

FYI: I recommend reviewing this PR commit-by-commit just to make sure that every dependency and plugin has been migrated appropriately, just to make sure that there isn't any left-overs that need addressing.

PS: The [gradle/libs.versions.toml](https://github.com/wordpress-mobile/WordPress-Android/pull/21262/files#diff-697f70cdd88ba88fe77eebda60c7e143f6ad1286bca75017421e93ad84fb87df) file, on both, its `versions` and `libraries` sections:
- It is alphabetically ordered.
- It is using the [kebal-case](https://en.wikipedia.org/wiki/Letter_case#Kebab_case) naming convention (or else `dash-case`) .
- All words are lowercased.
- Where it makes sense the `org` name is added in its prefix to group dependencies together and make them easily identifiable.
- Where needed the word `main` is added in its suffix to make sure that a dependency cannot be a parent of another dependency and a dependency itself. This helps in various corner cases.

-----

## To Test:



- Smoke test both, the `Jetpack` and `WordPress` apps, and verify everything is working as expected.
- You could also run the `./gradlew buildHealth` task locally and verify that everything is working as expected ([CI](https://buildkite.com/automattic/wordpress-android/builds/19907)).

-----

## Regression Notes

1. Potential unintended areas of impact

    - Not that I can think of.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - See `To test` section above.

3. What automated tests I added (or what prevented me from doing so)

    - N/A

-----

## PR Submission Checklist:


-----

## Testing Checklist (strike-out the not-applying and unnecessary ones): `N/A`
